defaults_test.go

```diff
@@ -44,10 +44,12 @@ type Sample struct {
 	Float64   float64       `default:"1.64"`
 	BoolTrue  bool          `default:"true"`
 	BoolFalse bool          `default:"false"`
+	BoolPtr   *bool         `default:"true"`
 	String    string        `default:"hello"`
 	Duration  time.Duration `default:"10s"`
 
 	IntOct    int    `default:"0o1"`
+	IntOctPtr *int   `default:"0o1"`
```

Comment:
Since its default value doesn't need to be an octal integer notation, we would like to add a more generic field like `IntPtr` instead.

```diff
	Int       int           `default:"1"`
+	IntPtr    *int          `default:"1"`
...
	IntOct    int    `default:"0o1"`
-	IntOctPtr *int   `default:"0o1"`
```

Also, I'd appreciate it if could you add test cases for pointers of other primitive types (namely `*unit`, `*float32`, and `*string`) and those of a slice (`*[]int`) and a map (`*map[string]int`).